packages/java-edition/src/mcfunction/index.ts

```diff
@@ -70,6 +71,84 @@ export const initialize = (
 		),
 	)
 
+	mcdoc.runtime.registerAttribute(
+		meta,
+		'command_argument',
+		mcdoc.runtime.attribute.validator.map(
+			mcdoc.runtime.attribute.validator.string,
+			core.ResourceLocation.lengthen,
+		),
+		{
+			attachString: (config, ctx) => {
+				let argParser
+				try {
+					argParser = parser.argument({
+						type: 'argument',
+						parser: config,
+					})
+				} catch (e) {
+					ctx.logger.warn(
+						`[mcdoc command_argument] Failed to create parser ${config}`,
+					)
+					return undefined
+				}
+				return (node) => {
+					const src = new core.Source(node.value, node.valueMap)
+					if (!argParser) {
+						ctx.err.report(
+							localize(
+								'mcfunction.parser.unknown-parser',
+								localeQuote(config),
+							),
+							core.Range.create(src.cursor, src.skipRemaining()),
+							core.ErrorSeverity.Hint,
+						)
+						return
+					}
+					const child = argParser(src, ctx)
+					if (child === core.Failure) {
+						ctx.err.report(
+							localize('mcfunction.parser.eoc-unexpected'),
+							node,
+						)
+						return
+					}
+					if (src.canRead()) {
+						ctx.err.report(
+							localize('mcdoc.runtime.checker.trailing'),
+							core.Range.create(src.cursor, src.skipRemaining()),
+						)
```

Comment:
I think it's bad design to report an error here.

Ideally, there should be a `filter` function defined which reports an appropriate error.

Was the intention of `internal` errors to facilitate sth like this with custom error reporting, and custom error condensing after my current PR is done? I could see that, as runtime can't really know the kinds of errors that can be reported.

I wouldn't call it `internal` I think, maybe more like `custom`, and ofc it shouldn't be created in runtime checker.

Reply:
I don't think it's bad design. The `internal` basically means "I as the attribute will handle errors in this type, but if you ever need to decide between multiple types, you can throw me away"

For complicated parsers it's much nicer to let them report their own errors, rather than them all having to go through the mcdoc runtime checker.

Reply:
Well, internal errors right now are only reported by the id attribute right now, so there actually is no filtering at all for any of the other complex parsers.

> For complicated parsers it's much nicer to let them report their own errors, rather than them all having to go through the mcdoc runtime checker.

I think, ideally all the parsing would happen when intitially encountering the node already, including reporting custom errors (as the parser would do normally, as nice as possible, not just a generic thing for the entire node). For error condensing, custom errors could be handled like you are doing it with internal right now, it would just be a bit tricky if we have comflicting custom errors from multiple definitions, but at this point we could just report a generic `invalid value` or something.

If you report errors here, and do the parsing as I suggested before, this implies to parse two times. So, there should be some way in error reporting to report custom errors. Maybe custom errors could just have a report function attached or something like that.

Reply:
I think the way Misode is doing it here is fine